### PR TITLE
Multi build

### PR DESCRIPTION
**Pull request by constantinius**
This PR deals with the various builds to be done (Build Matrix):

 - OS: Ubuntu, CentOS
 - Python: 2, 3
 - Django: 1.11, 2.2

Additional Tasks
 - Automatic building of Docker images and pushing to Docker hub
 - Automatic bundling of Python Packages and pushing to pypi

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,2 +1,6 @@
 .git
-autotest
\ No newline at end of file
+autotest
+build
+dist
+documentation
```
Comment (Schpidi) [inline .dockerignore]: This looks dangerous as this path is part of the repository isn't it?

File: .dockerignore [hunk 2]
```diff
@@ -1,2 +1,6 @@
...
+schemas
```
Comment (Schpidi) [inline .dockerignore]: Same again.

File: .dockerignore (hunk 1 above)
Comment (constantinius) [reply .dockerignore]: Sure, but these directories are not used in the docker image build process.

File: .dockerignore (hunk 2 above)
Comment (constantinius) [reply .dockerignore]: This references the same file as above.

**Pull request merged**